### PR TITLE
chore(web): use is_multi_provider_configuration_enabled feature flag

### PR DESCRIPTION
**Pull request by LetItRock**
### What change does this PR introduce?

Use the `IS_MULTI_PROVIDER_CONFIGURATION_ENABLED` feature flag to toggle between the Integrations List Page / Integrations Store Tiles.

For the e2e tests approach I think that we have to run for both pages, that's why I invented a way to achieve it with the global variable `_cypress`, check the code and comments.

### Other information (Screenshots)
Screenshot 2023-06-23 at 11 32 14
Screenshot 2023-06-23 at 11 32 23

### CODE REVIEW

File: apps/web/cypress/tests/integration-store.spec.ts [hunk 1]
```diff
@@ -1,6 +1,14 @@
+// @ts-nocheck
 import { ChannelTypeEnum, InAppProviderIdEnum } from '@novu/shared';
 
-describe.skip('Integration store page', function () {
+Cypress.on('window:before:load', (win) => {
+  win._cypress = {
+    ...win._cypress,
+    IS_INTEGRATIONS_LIST_PAGE_ENABLED: 'false',
+  };
+});
```
Comment (LetItRock) [inline apps/web/cypress/tests/integration-store.spec.ts:9]: Set the `_cypress.IS_INTEGRATIONS_LIST_PAGE_ENABLED` flag to false, it is then used in the configuration file where we do set up the env variables. 
I wasn't able to use the `window._env_` property as it overrides somewhere during the application initialization all the values.

File: apps/web/cypress/tests/integrations-list-page.spec.ts [hunk 2]
```diff
@@ -1,3 +1,12 @@
+// @ts-nocheck
+
+Cypress.on('window:before:load', (win) => {
+  win._cypress = {
+    ...win._cypress,
+    IS_INTEGRATIONS_LIST_PAGE_ENABLED: 'true',
+  };
+});
```
Comment (LetItRock) [inline apps/web/cypress/tests/integrations-list-page.spec.ts:8]: similar approach with flag `true`, this way we can run both e2e tests for these pages

File: apps/web/src/config/index.ts [hunk 3]
```diff
@@ -61,7 +61,8 @@ export const IS_TEMPLATE_STORE_ENABLED = isCypress
   : window._env_.IS_TEMPLATE_STORE_ENABLED || process.env.IS_TEMPLATE_STORE_ENABLED || 'false';
 
 export const IS_MULTI_PROVIDER_CONFIGURATION_ENABLED = isCypress
-  ? window._env_.IS_MULTI_PROVIDER_CONFIGURATION_ENABLED ||
+  ? window._cypress.IS_MULTI_PROVIDER_CONFIGURATION_ENABLED ||
```
Comment (LetItRock) [inline apps/web/src/config/index.ts:64]: use the flag

**Review by p-fernandez — APPROVED** [review]
🌟

File: apps/web/src/config/index.ts (hunk 3 above)
Comment (djabarovgeorge) [reply apps/web/src/config/index.ts:64]: Those config conditions are becoming more and more complex to understand at first glance, maybe we could do another approach in the near future. (not related to this PR :) )

File: apps/web/src/pages/integrations/IntegrationsListPage.tsx [hunk 4]
```diff
@@ -104,5 +105,7 @@ const IntegrationsList = () => {
 };
 
 export const IntegrationsListPage = () => {
-  return <IntegrationsList />;
+  const isIntegrationsListPageEnabled = useIsMultiProviderConfigurationEnabled();
+
+  return isIntegrationsListPageEnabled ? <IntegrationsList /> : <IntegrationsStore />;
```
Comment (djabarovgeorge) [inline apps/web/src/pages/integrations/IntegrationsListPage.tsx:110]: 👌

**Review by djabarovgeorge — APPROVED** [review]
🚀

**Pull request merged**